lib/admin.php

```diff
@@ -234,100 +240,134 @@ function category_select( $option, $description ) {
     }
 
     $categories = $remote['category_list']['categories'];
-    $selected = isset( $options['publish-category'] ) ? $options['publish-category'] : '';
+    return $categories;
+  }
 
-    echo "<select id='discourse[{$option}]' name='discourse[{$option}]'>";
-    echo '<option></option>';
+  function category_select( $option, $description ) {
+    $options = get_option( 'discourse' );
 
-    foreach( $categories as $category ){
-      printf( '<option value="%s"%s>%s</option>',
-        $category['id'],
-        selected( $selected, $category['id'], false ),
-        $category['name']
-      );
-    }
+    $force_update = isset($options['publish-category-update']) ? $options['publish-category-update'] : '0';
+    $categories = self::get_discourse_categories($force_update);
 
-    echo '</select>';
-  }
+    if( is_wp_error( $categories ) ) {
+     self::text_input( $option, $description );
+     return;
+   }
 
-  function text_input( $option, $description ) {
-    $options = $this->options;
+   $selected = isset( $options['publish-category'] ) ? $options['publish-category'] : '';
+   $name = "discourse[{$option}]";
+   self::option_input($name, $categories, $selected);
+ }
 
-    if ( array_key_exists( $option, $options ) ) {
-      $value = $options[$option];
-    } else {
-      $value = '';
-    }
+ function option_input($name, $group, $selected){
+   echo "<select id='$name' name='$name'>";
```

Comment:
inconsistent quotes, but you can leave for now as there's other instances like this in the existing codebase